core_lib/src/qminiz.cpp

```diff
@@ -166,17 +165,22 @@ Status MiniZ::uncompressFolder(QString zipFilePath, QString destPath)
     baseDir.makeAbsolute();
 
     mz_zip_archive* mz = new mz_zip_archive;
+    ScopeGuard mzScopeGuard([&] {
+        delete mz;
+    });
+
     mz_zip_zero_struct(mz);
 
     mz_bool ok = mz_zip_reader_init_file(mz, zipFilePath.toUtf8().data(), 0);
-
-    ScopeGuard mzScopeGuard([&] {
+    ScopeGuard mzScopeGuard2([&] {
```

Comment:
You can combine `mzScopeGuard` and `mzScopeGuard2` into a single one.

Reply:
Indeed, I just prefer having two different scope guards in this case, mzScopeGuard2 is only relevant if the writer has been opened, while the first one is always relevant.

Reply:
I was concerned about the destruction order of local objects, which can be a bit tricky. Then I checked the C++ standards, local objects are destructed in reverse order of constructions, so `mzScopeGuard2` will always execute before `mzScopeGuard`. This should be fine.